### PR TITLE
Add cassandra role

### PR DESCRIPTION
**Pull request by maxrothman**
Submitting for initial review

Uses JBOD, rather than RAID0 or lvm. It should be pretty simple to integrate either of those once the config code for them is written.

Default vars support single-instance deployment, but comments suggest alternatives for multi-node deployments.

Currently does not support authentication, because doing so is a little more complicated (involves creating a keyspace and changing its replication factor).

### CODE REVIEW

**Comment by maxrothman** [comment]
Ready for review @e0d @jibsheet @feanil @fredsmith

File: playbooks/roles/cassandra/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# cassandra.yaml basic configuration parameters
+cassandra_cluster_name: "Test Cluster"
+
+#Set this!
+cassandra_seeds: ["127.0.0.1"]
+
+#should eventually use EC2Snitch
+cassandra_snitch: "SimpleSnitch"
+
+#set this ONLY when initializing a new cluster with NO DATA
```
Comment (e0d) [inline playbooks/roles/cassandra/defaults/main.yml:28]: Given the warning, should this default to false?  I'd rather have new clusters be the exception than than to risk data loss.

File: playbooks/roles/cassandra/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
+
+- name: Make sure the cassandra group is present
+  group:
+    name: "{{cassandra_group}}"
+    state: present
+
+- name: Make sure the cassandra user is present
```
Comment (e0d) [inline playbooks/roles/cassandra/tasks/main.yml]: Isn't the package creating users?

File: playbooks/roles/cassandra/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
+
+- name: Make sure the cassandra group is present
+  group:
+    name: "{{cassandra_group}}"
```
Comment (e0d) [inline playbooks/roles/cassandra/tasks/main.yml]: white space, {{ cassandra_group }}

**Comment by e0d** [comment]
This role should contain tasks for creating users in Cassandra or we should have a separate play like we do for MySQL, either way we need to ansible to do it and the dictionary that specifies users.

File: playbooks/roles/cassandra/defaults/main.yml (hunk 1 above)
Comment (maxrothman) [inline playbooks/roles/cassandra/defaults/main.yml:28]: Good point.

**Comment by maxrothman** [comment]
@e0d turns out the package does create the users. I've removed all of that logic and added a note to the variables warning people not to change them.

File: playbooks/roles/cassandra/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+
+cassandra_version: "2.0.14"
+cassandra_source_url: "http://archive.apache.org/dist/cassandra/{{cassandra_version}}/apache-cassandra-{{cassandra_version}}-bin.tar.gz"
```
Comment (feanil) [inline playbooks/roles/cassandra/defaults/main.yml]: Looks like we don't use this source?

File: playbooks/roles/cassandra/templates/cassandra-env.sh.j2 [hunk 5]
```diff
@@ -0,0 +1,281 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (feanil) [inline playbooks/roles/cassandra/templates/cassandra-env.sh.j2]: Looks like there are no changes in this file, do we need to override it?

**Comment by feanil** [comment]
:+1:  once the above two comments are resolved.

File: playbooks/roles/cassandra/defaults/main.yml (hunk 4 above)
Comment (e0d) [inline playbooks/roles/cassandra/defaults/main.yml]: If we need it fix the whitespace.

**Comment by fredsmith** [comment]
LGTM :+1:

**Comment by fredsmith** [comment]
:shipit:

**Comment by feanil** [comment]
:shipit:   :anchor:  away!

**Pull request merged**